src/admissioncontrollerpolicy/admissionControllerPolicy.go

```diff
@@ -62,26 +62,46 @@ func DeleteKyvernoPolicies(kyvernoPolicies []string, namespace string, labels ma
 }
 
 // UpdateOrInsertPolicyYamlToDB upserts admission controller policies to DB
-func UpdateOrInsertPolicyYamlToDB(kyvernoPolicies []kyvernov1.Policy, labels types.LabelMap) {
+func UpdateOrInsertPolicyYamlToDB(kyvernoPolicies []kyvernov1.PolicyInterface, labels types.LabelMap, generic bool) {
 
 	var res []types.PolicyYaml
-	for _, kyvernoPolicy := range kyvernoPolicies {
-		jsonBytes, err := json.Marshal(kyvernoPolicy)
-		if err != nil {
-			log.Error().Msg(err.Error())
-			continue
+	for _, kyvernoPolicyInterface := range kyvernoPolicies {
+		var jsonBytes []byte
+		var yamlBytes []byte
+		var err error
+
+		switch kyvernoPolicyInterface.(type) {
+		case *kyvernov1.ClusterPolicy:
```

Comment:
The particular loc you mentioned don't have anything to do with recommendation. They just put the policy into the database.

> Currently, the way I read it, it will be recommended in context to every workload?

Cluster Policies are applicable to the complete cluster, they are workload independent.